src/acceptance/assets/app/go_app/internal/app/cpu_test.go

```diff
@@ -63,13 +64,24 @@ var _ = Describe("CPU tests", func() {
 
 					By("wasting cpu time")
 					cpuWaster := &app.ConcurrentBusyLoopCPUWaster{}
+
 					cpuWaster.UseCPU(utilisation, duration)
+
 					Expect(cpuWaster.IsRunning()).To(Equal(true))
 					Eventually(cpuWaster.IsRunning).WithTimeout(duration + time.Second).WithPolling(time.Second).Should(Equal(false))
+
 					newCpu := getTotalCPUUsage("after test")
+
 					expectedCPUUsage := multiplyDurationByPercentage(duration, utilisation)
-					// Give 10% tolerance - but at least 1 second, as this is the internal resolution of the CPU waster
-					tolerance := max(multiplyDurationByPercentage(expectedCPUUsage, 10), time.Second)
+
+					// If the environment variable CI is not set to true: Give 10% tolerance - but at least 1 second, as this is the internal resolution of the CPU waster.
+					// If the environment variable CI is set to true, as is by default in GitHub Actions, give 50% tolerance. This is due to the fact that on CI workers the available CPU time is not guaranteed.
```

Comment:
Is there an official documentation or something related to this that we could link to?

Reply:
Yes! https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

Reply:
```suggestion
					// If the environment variable CI is set to true, as is by default in GitHub Actions 
					// (see https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)
					// give 50% tolerance. This is due to the fact that on CI workers the available CPU time is not guaranteed.
```

Reply:
> This is due to the fact that on CI workers the available CPU time is not guaranteed.

Let's don't document the boring `CI: true` :D My comment was more related to not guaranteed CPU time

Reply:
Unfortunately, I didn't find this documented on GitHub. :disappointed:

Reply:
No problem, thanks for checking!

Reply:
I've updated the PR description to make note of it:

> The working assumption is that this due to GitHub Action workflows share resources with other workflow executions if ran on GitHub-hosted runners.